tests/TestHarness/transactions.py

```diff
@@ -152,48 +152,60 @@ def transferFundsAsync(self, source, destination, amountStr, memo="memo", force=
         return popen, cmdArr
 
     # publish contract and return transaction as json object
-    def publishContract(self, account, contractDir, wasmFile, abiFile, waitForTransBlock=False, shouldFail=False, sign=False):
+    def publishContract(self, account, contractDir, wasmFile, abiFile, waitForTransBlock=True, shouldFail=False, sign=False, retryNum:int=5):
+        assert(isinstance(retryNum, int))
         signStr = NodeosQueries.sign_str(sign, [ account.activePublicKey ])
-        cmd="%s %s -v set contract -j %s %s %s" % (Utils.EosClientPath, self.eosClientArgs(), signStr, account.name, contractDir)
+        cmd=f"{Utils.EosClientPath} {self.eosClientArgs()} -v set contract -j -f {signStr} {account.name} {contractDir}"
         cmd += "" if wasmFile is None else (" "+ wasmFile)
         cmd += "" if abiFile is None else (" " + abiFile)
         if Utils.Debug: Utils.Print("cmd: %s" % (cmd))
         trans=None
```

Comment:
I can't tell if it is needed, but I think we should move `trans=None` in the while loop.

Reply:
Not needed, but doesn't hurt. Moved into while.